add verification to check if escapeKeys have elements (pkg/term/proxy.go)

Sh4d1 opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/moby/moby/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

For additional information on our contributing process, read our contributing
guide https://docs.docker.com/opensource/code/

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

Fixes https://github.com/moby/moby/issues/36917

I think it does not impact the function if the array is not empty.

**- What I did**
Add a check to see if the array is not empty
**- How I did it**
Compare the length of array to 0
**- How to verify it**

**- Description for the changelog**
Add verification of escapeKeys array length in `pkg/term/proxy.go`
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->


**- A picture of a cute animal (not mandatory but encouraged)**
![cute cat](https://wallpapercave.com/wp/WDyRZJd.jpg)

GordonTheTurtle commented:
<!-- AUTOMATED:POULE:DCO-EXPLANATION -->
Please sign your commits following these rules:
https://github.com/moby/moby/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "master" git@github.com:Sh4d1/moby.git somewhere
$ cd somewhere
$ git commit --amend -s --no-edit
$ git push -f
~~~

Amending updates the existing PR. You **DO NOT** need to open a new one.

fntlnz commented:
@Sh4d1 thanks, can you add a test case in `TestEscapeProxyRead` related to this?

Sh4d1 commented:
@fntlnz  are the two test I added enough ?

fntlnz commented on pkg/term/proxy.go:
```diff
@@ -51,6 +51,10 @@ func (r *escapeProxy) Read(buf []byte) (int, error) {
 	}
 
+	if len(r.escapeKeys) == 0 {
+		return nr, nil
```
This is never reached, in order to raech this you will need to add a test case that actually has no escapeKeys and more only one key, cases with more than one key are already caught here: https://github.com/moby/moby/blob/111457538761414254a8c1a3eb2ac9654b36aff8/pkg/term/proxy.go#L47-L52

Sh4d1 commented:
My bad, I forgot that :p 
Changes pushed :)

vdemeester reviewed (approved):
SGTM :frog:

Sh4d1 commented:
@fntlnz done!

thaJeztah commented on pkg/term/proxy.go:
```diff
@@ -51,6 +51,10 @@ func (r *escapeProxy) Read(buf []byte) (int, error) {
 		return nr, err
 	}
 
+	if len(r.escapeKeys) == 0 {
```
Should this be more at the top of the function? (before the `preserve = func() {` ?)

Sh4d1 replied:
I guess it makes more sense, I'll push the changes quickly :)

Sh4d1 committed:
Add verification and test to check if escapeKeys is not empty

In pkg/term/proxy.go and pkg/term/proxy_test.go, check if escapeKeys is empty and if it is, return the one key read

Signed-off-by: Patrik Cyvoct <patrik@ptrk.io>

Sh4d1 commented:
@thaJeztah @dnephin more changes needed? Or is it good for you?

thaJeztah reviewed (approved):
LGTM
